core/src/main/scala-2.13+/cats/data/NonEmptyLazyList.scala

```diff
@@ -62,14 +62,46 @@ object NonEmptyLazyList extends NonEmptyLazyListInstances {
   def fromSeq[A](as: Seq[A]): Option[NonEmptyLazyList[A]] =
     if (as.nonEmpty) Option(create(LazyList.from(as))) else None
 
-  def fromLazyListPrepend[A](a: A, ca: LazyList[A]): NonEmptyLazyList[A] =
-    create(a +: ca)
+  def fromLazyListPrepend[A](a: => A, ll: => LazyList[A]): NonEmptyLazyList[A] =
+    create(a #:: ll)
 
-  def fromLazyListAppend[A](ca: LazyList[A], a: A): NonEmptyLazyList[A] =
-    create(ca :+ a)
+  def fromLazyListAppend[A](ll: => LazyList[A], a: => A): NonEmptyLazyList[A] =
+    create(ll #::: a #:: LazyList.empty)
 
   def apply[A](a: => A, as: A*): NonEmptyLazyList[A] =
-    create(LazyList.concat(LazyList(a), LazyList.from(as)))
+    create(a #:: LazyList.from(as))
+
+  // allows the creation of fully lazy `NonEmptyLazyList`s by prepending to this
+  def maybe[A](ll: => LazyList[A]): Maybe[A] = new Maybe(() => ll)
```

Comment:
IMO, this method requires a way more comprehensive scaladocs (preferably with examples).

Reply:
haven't written the docs yet, lol; not gonna do that until the design is solid enough to move forward, so I don't waste effort (docs are hard).

that said, the explanation is fairly simple: `fromLazyList` and `fromLazyListUnsafe` are bad, because they evaluate the first element of the `LazyList`. to remedy that, `Maybe` allows you to prepend either single elements or `NonEmptyLazyList`s, guaranteeing that the result is not empty without evaluating any elements

Reply:
trivial usage example:

```scala
val nell: NonEmptyLazyList[Int[] = 1 #:: NonEmptyLazyList.maybe(LazyList.empty)
```

Reply:
better now?

Reply:
@NthPortal to be honest, I'm still not getting what the purpose those `Maybe` and `Deferred` classes are supposed to serve for. IMO by design `LazyList` is a collection that lazily evaluates its items upon the first traversal. But the instance of `LazyList` is not supposed to be lazy itself. Therefore it would be nice to have examples that would show how exactly those classes can be useful. And why the same functionality cannot be achieved without them.

Reply:
Oh, I see that `Deferrer` is a similar thing to one in the std `LazyList`, so it seems to be fine then. But `Deferrer` is just a syntax that is added implicitly and hence transparently for users. Which is not the case for `Maybe`.

Reply:
@satorg I'll look into adding an example or two to the docs of `maybe`